app/src/ui/lib/tooltipped-commit-sha.tsx

```diff
@@ -0,0 +1,51 @@
+import { clipboard } from 'electron'
+import React from 'react'
+import { TooltipDirection } from './tooltip'
+import { TooltippedContent } from './tooltipped-content'
+
+interface ITooltippedCommitSHAProps {
+  readonly className?: string
+  /**
+   * Short SHA to display in the tooltipped element, or the tooltipped element
+   * itself.
+   */
+  readonly shortSHA: string | JSX.Element
+
+  /** Long SHA to display within the tooltip. */
+  readonly longSHA: string
```

Comment:
Just a thought since this is a formatting component -> feel free to ignore. We could pass in just longSHA: string | Commit. If is a string, we shorten it here using shortenSha method and if it is a commit we use commit.shortSha and commit.sha.

Reply:
(Or make longSHA? and commit? properties and use the defined one :D)

Reply:
Well, the problem is the short SHA is either a `string` or a `JSX.Element`, if you need something custom (like we do in the submodule interstitial to use a `Ref` component to wrap the SHA).

Reply:
That could be done either as another component extending this or as boolean `asRef` and still keep all the formatting to his component or components.